alts/build.gradle

```diff
@@ -29,7 +28,14 @@ dependencies {
         exclude group: 'io.grpc', module: 'grpc-context'
     }
     compileOnly libraries.javax_annotation
-    testCompile project(':grpc-testing'),
+
+    // Use grpc-netty-shaded for transitive dependency.
+    compileOnly project(':grpc-netty')
+    shadow project(path: ':grpc-netty-shaded', configuration: 'shadow')
+    shadow configurations.compile.getDependencies()
```

Comment:
This is doubly suspicious. It is strange to need to copy the dependencies, but wouldn't this also include compile-only dependencies?

Reply:
My understanding is:

 * `DependencyHandler.project(path : 'grpc-alts', configuration: 'shadow')` returns a `Dependency` that includes the output of `:grpc-alts:shadowJar` plus all dependencies added (manually) into `shadow` configuration.

However,

* The output of `:grpc-alts:shadowJar` does not depend on anything in the shadow configuration.

* The output pom.xml of maven publication does not include anything in the shadow configuration unless you modify `pom.withXml {}` manually.